update-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/ApiUpdateDoiRequestHandlerTest.java

```diff
@@ -0,0 +1,351 @@
+            publication.getIdentifier(),
+            publication.getOwner(),
+            includedDoiRequest,
+            publication.getEntityDescription());
+    }
+
+    private Publication insertPublicationWithDoiRequest(Clock clock)
+        throws com.fasterxml.jackson.core.JsonProcessingException {
+        Publication publication = PublicationGenerator.getPublicationWithDoiRequest(clock)
+            .copy()
+            .withCreatedDate(mockOneHourBefore)
+            .build();
+        insertPublication(publicationsTableName, publication);
+        return publication;
+    }
+
+    private Publication insertPublicationWithoutDoiRequest(Clock clock)
+        throws com.fasterxml.jackson.core.JsonProcessingException {
+        Publication publication = PublicationGenerator.getPublicationWithoutDoiRequest(clock)
+            .copy()
+            .withCreatedDate(mockOneHourBefore)
+            .build();
+        insertPublication(publicationsTableName, publication);
+        return publication;
+    }
+
+    private ApiUpdateDoiRequest requestWithoutPublicationId() {
+        return new ApiUpdateDoiRequest();
+    }
+
+    private <T> GatewayResponse<T> sendRequest(ApiUpdateDoiRequest doiRequest, String username) throws IOException {
+        var pathParams = Map.of(API_PUBLICATION_PATH_IDENTIFIER, doiRequest.getPublicationId());
+        InputStream input = createRequest(doiRequest, username, pathParams);
+        ByteArrayOutputStream output = outputStream();
+        handler.handleRequest(input, output, context);
+
+        return GatewayResponse.fromOutputStream(output);
+    }
+
+    private <T> GatewayResponse<T> sendRequest(ApiUpdateDoiRequest doiRequest, JsonNode requestContext)
+        throws IOException {
+        var pathParams = Map.of(API_PUBLICATION_PATH_IDENTIFIER, doiRequest.getPublicationId());
+        InputStream input = createRequest(doiRequest, pathParams, requestContext);
+        ByteArrayOutputStream output = outputStream();
+        handler.handleRequest(input, output, context);
+
+        return GatewayResponse.fromOutputStream(output);
+    }
+
+    private <T> GatewayResponse<T> sendRequest(ApiUpdateDoiRequest doiRequest,
+                                               String username,
+                                               Map<String, String> pathParameters) throws IOException {
+        InputStream input = createRequest(doiRequest, username, pathParameters);
+        ByteArrayOutputStream output = outputStream();
+        handler.handleRequest(input, output, context);
+
+        return GatewayResponse.fromOutputStream(output);
+    }
+
+    private InputStream createRequest(ApiUpdateDoiRequest doiRequest,
+                                      String username,
+                                      Map<String, String> pathParameters)
+        throws JsonProcessingException {
+        var requestContext = objectMapper.createObjectNode();
+        requestContext
+            .putObject("authorizer")
```

Comment:
We use these a lot in testing…I wonder if they shouldn't be put somewhere as constants :)

Reply:
Yes, it belongs inside RequestInfo object, should do a separate task on this as we want to fix this across all code. 
(mantra: tell objects what to do instead of asking them about information)